lib/main.js

```diff
@@ -90,6 +148,44 @@ module.exports = class Router extends EventEmitter {
         this.app.get('/bundle/:file', this.getFileCallback());
         this.app.get('/test/bundle/:file', this.getTestFileCallback());
 
+        this.app.post('/publish-assets', async (req, res, next) => {
+            try {
+                const payload = await getBody(req);
+                const bundler = new OptimisticBundler({
+                    env: this.options.env,
+                    minify:
+                        req.query.minify === 'true' ||
+                        this.options.isProduction,
+                    sink: this.sink,
+                    sourceMaps: req.query.sourceMaps === 'true',
+                    logger: this.log || emitLogger(this),
+                });
+                const meta = await bundler.publishAssets(payload);
+                res.send(meta);
+            } catch (err) {
+                next(err);
+            }
+        });
+
+        this.app.post('/publish-instructions', async (req, res, next) => {
+            try {
+                const payload = await parseBody(req, res);
+                const bundler = new OptimisticBundler({
```

Comment:
seems more reasonable to create a `new OptimisticBundler` in the constructor with `env`, `sink` `logger` and invoke `publishBlah` with the other options in these handlers

Reply:
hmm, yea good point. Rats, another refactor and test before merge I guess.